test/onnx/verify_onnx.cpp

```diff
@@ -1080,6 +1080,72 @@ TEST_CASE(nonzero_test)
     EXPECT(migraphx::verify_range(result_vector, gold));
 }
 
+TEST_CASE(nms_sanity_compute_test)
+{
+    migraphx::program p = migraphx::parse_onnx("nms_test.onnx");
+    p.compile(migraphx::ref::target{});
+
+    migraphx::shape mob_shape{migraphx::shape::int64_type, {1}};
+    std::vector<uint64_t> mob_data = {0};
+
+    migraphx::shape boxes_shape{migraphx::shape::float_type, {1, 6, 4}};
+    std::vector<float> boxes_data = {1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                                     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
+
+    migraphx::shape score_shape{migraphx::shape::float_type, {1, 1, 6}};
+    std::vector<float> score_data = {1, 2, 3, 4, 5, 6};
+
+    migraphx::shape score_thresh_shape{migraphx::shape::float_type, {1}};
+    std::vector<float> score_thresh = {0.0f};
+
+    migraphx::shape iou_thresh_shape{migraphx::shape::float_type, {1}};
+    std::vector<float> iou_thresh = {0.0f};
+
+    migraphx::parameter_map pp;
+    pp["max_output_boxes_per_class"] = migraphx::argument(mob_shape, mob_data.data());
+    pp["boxes"]                      = migraphx::argument(boxes_shape, boxes_data.data());
+    pp["scores"]                     = migraphx::argument(score_shape, score_data.data());
+    pp["score_threshold"]            = migraphx::argument(score_thresh_shape, score_thresh.data());
+    pp["iou_threshold"]              = migraphx::argument(iou_thresh_shape, iou_thresh.data());
+
+    p.eval(pp).back();
+
+    EXPECT(true);
+}
+
+TEST_CASE(nms_large_input_boxes_test)
+{
+    migraphx::program p = migraphx::parse_onnx("nms_test.onnx");
+    p.compile(migraphx::ref::target{});
+
+    migraphx::shape mob_shape{migraphx::shape::int64_type, {1}};
+    std::vector<uint64_t> mob_data = {9223372036854775807};
+
+    migraphx::shape boxes_shape{migraphx::shape::float_type, {1, 6, 4}};
+    std::vector<float> boxes_data = {1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                                     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
+
+    migraphx::shape score_shape{migraphx::shape::float_type, {1, 1, 6}};
+    std::vector<float> score_data = {1, 2, 3, 4, 5, 6};
+
+    migraphx::shape score_thresh_shape{migraphx::shape::float_type, {1}};
+    std::vector<float> score_thresh = {0.0f};
+
+    migraphx::shape iou_thresh_shape{migraphx::shape::float_type, {1}};
+    std::vector<float> iou_thresh = {0.0f};
+
+    migraphx::parameter_map pp;
+    pp["max_output_boxes_per_class"] = migraphx::argument(mob_shape, mob_data.data());
+    pp["boxes"]                      = migraphx::argument(boxes_shape, boxes_data.data());
+    pp["scores"]                     = migraphx::argument(score_shape, score_data.data());
+    pp["score_threshold"]            = migraphx::argument(score_thresh_shape, score_thresh.data());
+    pp["iou_threshold"]              = migraphx::argument(iou_thresh_shape, iou_thresh.data());
+
+    p.eval(pp).back();
+
+    EXPECT(true);
+}
```

Comment:
I think this would be better tested in ref_ops_test.cpp. Since you're checking the nonmaxsuppression.hpp compute function

Reply:
Sure I can move that